code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/integration/azure/AzureIdentityProvider.java

```diff
@@ -0,0 +1,202 @@
+                    authToken = authTokenDao.createToken(jwt, AzureConstants.CONFIG, account.getId());
+                    jwt = authToken.getKey();
+                } catch (ClientVisibleException e) {
+                    if (e.getCode().equalsIgnoreCase(AzureConstants.AZURE_ERROR) &&
+                            !e.getDetail().contains("401")) {
+                        throw new ClientVisibleException(ResponseCodes.INTERNAL_SERVER_ERROR,
+                                AzureConstants.JWT_CREATION_FAILED, "", null);
+                    }
+                }
+            } else {
+                jwt = authToken.getKey();
+            }
+
+        }
+        if (StringUtils.isBlank(jwt)){
+            return Collections.emptySet();
+        }
+        ApiRequest request = ApiContext.getContext().getApiRequest();
+        request.setAttribute(AzureConstants.AZURE_JWT, jwt);
+        request.setAttribute(AzureConstants.AZURE_ACCESS_TOKEN, accessToken);
+        request.setAttribute(AzureConstants.AZURE_REFRESH_TOKEN, refreshToken);
+        return azureTokenUtils.getIdentities();
+    }
+
+    private List<Identity> searchGroups(String groupName, boolean exactMatch) {
+        List<Identity> identities = new ArrayList<>();
+
+        AzureUserInfo group;
+        try {
+            group =  azureClient.getAzureGroupByName(groupName);
+            if (group == null){
+                return identities;
+            }
+        } catch (ClientVisibleException e) {
+            return identities;
+        }
+        Identity identity = group.toIdentity(AzureConstants.GROUP_SCOPE);
+        identities.add(identity);
+        return identities;
+
+    }
+
+    private List<Identity> searchUsers(String userName, boolean exactMatch) {
+        List<Identity> identities = new ArrayList<>();
+
+        AzureUserInfo user;
+        try {
+            user =  azureClient.getAzureUserByName(userName);
+        } catch (ClientVisibleException e) {
+            return identities;
+        }
+        if (user == null) {
+            return identities;
+        }
+        Identity identity = user.toIdentity(AzureConstants.USER_SCOPE);
+        identities.add(identity);
+        return identities;
+
+    }
+
+    @Override
+    public Identity getIdentity(String id, String scope) {
+        if (!isConfigured()){
+            notConfigured();
+        }
+        switch (scope) {
+            case AzureConstants.USER_SCOPE:
+                AzureUserInfo user = azureClient.getUserById(id);
+                return user.toIdentity(AzureConstants.USER_SCOPE);
+            case AzureConstants.GROUP_SCOPE:
+                AzureUserInfo org = azureClient.getGroupById(id);
+                return org.toIdentity(AzureConstants.GROUP_SCOPE);
+            default:
+                throw new ClientVisibleException(ResponseCodes.BAD_REQUEST,
+                        IdentityConstants.INVALID_TYPE, "Invalid scope for GithubSearchProvider", null);
```

Comment:
We should change the log message; looks like it referencing incorrect provider - github